app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -773,10 +810,30 @@ public boolean onBackPressed() {
             return true; // no code here, as previous() was used in the if
         }
 
-        // That means that we are on the start of the stack,
+        // If we passed previous check at the top `player.getPlayQueue().previous()`
+        // then we can be sure that queue have no more streams in the history.
+        // If the user chose BACK_BEHAVIOR_PREV_PLAYLIST behavior
+        // we have to minimize the player into the mini player here
+        if (behavior == PlayerHelper.BackBehavior.BACK_BEHAVIOR_PREV_PLAYLIST) {
+            bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+            return true;
```

Comment:
I still do not understand why this code would cause the previous item from the playlist to be played. Also, how do you check if a play queue item belongs to a playlist, and if its previous item also belongs to a playlist? And where is the code that queues the previous item from the playlist?

Or maybe I didn't understand what you mean by playlist, do you mean play queue (though still the code would not explain that, either)?

Reply:
Yes, I mean PlayQueue that has a history of played items. 
See https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java#L45

It's a List that holds previously played items from this queue. 

Here https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java#L772
I call this method https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java#L506
which returns true in case it started to play previously played stream. If it isn't then it returns false and the body of `if` at https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java#L772 will be skipped.

Maybe would be better to rename `previous()` method to something like `playPreviousFromHistoryIfAny()` or something like that. Or maybe make two methods insteaf of `previous()` like `hasPrevious()` and `playPrevious()`. But I think having one method is better because calling `playPrevious` without checking `hasPrevious` (when no previous) would raise an exception which is not needed at all.